docs/source/configuration/authorization.mdx

```diff
@@ -98,199 +138,263 @@ So the `me` top level operation would not even be executed. When returning the r
 
-As an example, assuming we have these subgraphs definition:
+Consider these example subgraph schemas:
 
-```graphql
-# subgraph A
+```graphql title="Product subgraph"
 type Query {
-  t: T
+  product: Product
 }
 
-type T @key(fields: "id") {
+type Product @key(fields: "id") {
   id: ID! @authenticated
+  name: String!
+  price: Int @authenticated
 }
 ```
 
-```graphql
-# subgraph B
-type T @key(fields: "id") {
+```graphql title="Inventory subgraph"
+type Query {
+  product: Product
+}
+
+type Product @key(fields: "id") {
   id: ID! @authenticated
-  value: Int
+  inStock: Boolean!
 }
 ```
 
-The following query can be done when not authenticated:
+An unauthenticated request would successfully execute this query:
 
 ```graphql
-{
-  t {
-    value
+query {
+  product {
+    name
+    inStock
   }
 }
 ```
 
-But this one would end up with a `null` in `id`, which would trigger nullability rules and nullify `t` too.
+But for this query:
 
 ```graphql
-{
-  t {
+query {
+  product {
     id
-    value
+    name
   }
 }
 ```
 
-### Interfaces
+an unauthenticated request would resolve `null` for `id`. And since `id` is a non-nullable field, `product` would return `null`.
+
+This behavior is similar to what you can create with [contracts](/graphos/delivery/contracts/) and the [`@inaccessible` directive](https://www.apollographql.com/docs/federation/federated-types/federated-directives/#inaccessible).
 
-In the case one of the types implementing an interface requires authorization, then querying the interface will be allowed, but any parts that require access to that type will be filtered.
+### Authorization and interfaces
 
-As an example, with this schema, where the interface  `I` does not require authentication, but the `User` type does:
+If a type [implementing an interface](https://www.apollographql.com/docs/apollo-server/schema/unions-interfaces/#interface-type) requires authorization, unauthorized requests can query the interface, but not any parts of the type that require authorization.
+
+For example, consider this schema where the `User` interface  doesn't require authentication, but the `Admin` type which implements `User` does:
 
 ```graphql
 type Query {
-  itf: I!
+  users: [User!]!
 }
 
-interface I {
-    id: ID
+interface User {
+  id: ID
+  name: String
 }
 
-type User
-implements I
+type Admin
+implements User
 @authenticated {
   id: ID
   name: String
+  role: String 
 }
 ```
 
-If we send this query with an inline fragment and a type condition on `User`:
+If an unauthenticated request were to make this query:
 
 ```graphql
 query {
-    itf {
-        id
-        ... on User {
-          name
-        }
+  users {
+    id
+    name
+    ... on Admin {
+      role
     }
+  }
 }
 ```
 
-The query would be filtered as follows:
+The router woudl filter the query as follows:
 
 ```graphql
 query {
-  itf {
+  users {
     id
```

Comment:
@Geal  This confuses me... wouldn't `id` be restricted since the entire `User` type is restricted?

Reply:
@lleadbet , any insight on this?

Reply:
Closing this loop, but the Slack thread I nudged you in covers this I believe. Still sorting out which is "correct" but the above is correct as of now.